hw_7.py

```diff
@@ -0,0 +1,125 @@
+# just debug print
+print("Computer generated a number for you. Try to guess!")
+
+
+def player_number_check():
+    """Проверка ответа пользователя на уникальность цифр"""
+    player_number = int(input("Enter your number "
+                              "(should contain only uniq digits): "))
+    player_number_list = list(str(player_number))
+    if len(set(player_number_list)) == 4:
+        return player_number
+    return player_number_check()
+
+
+def bulls_cows_counter(computer_number, player_number, bulls=0, cows=0):
+    """Подсчет быков и коров"""
+    computer_number_list = list(str(computer_number))
+    player_number_list = list(str(player_number))
+    for n in range(len(player_number_list)):
+        if player_number_list[n] == computer_number_list[n]:
+            bulls += 1
+        elif player_number_list[n] in computer_number_list:
+            cows += 1
+    return bulls, cows
+
+
+# 1-й вызов функции проверки ввденного игроком числа
+pl_num_attempt = player_number_check()
+# 1-й вызов функции подсчета быков и коров
+bulls_guess, cows_guess = bulls_cows_counter(comp_numb_gen, pl_num_attempt)
+print("You guessed", bulls_guess, "bull(s) and", cows_guess, "cow(s).")
+
+# Цикл повтора вышеуказанных функций пока количество быков не станет == 4
+while bulls_cows_counter(comp_numb_gen, pl_num_attempt) != (4, 0):
+    print("    Try one more time!")
+    pl_num_attempt = player_number_check()
+    bulls_guess, cows_guess = bulls_cows_counter(comp_numb_gen, pl_num_attempt)
+    print("You guessed", bulls_guess, "bull(s) and", cows_guess, "cow(s).")
+print("Congratulation!", pl_num_attempt, "is the correct answer :)")
+
+
+# # # Пирамида # # #
+# Мы можем визуализировать художественную пирамиду ASCII с N уровнями,
+# напечатав N рядов звездочек,
+# где верхний ряд имеет одну звездочку в центре,
+# а каждый последующий ряд имеет две дополнительные звездочки с каждой стороны.
+# Вот как это выглядит, когда N равно 3.
+#   *
+#  ***
+# *****
+# Вот как это выглядит, когда N равно 5.
+#     *
+#    ***
+#   *****
+#  *******
+# *********
+# Необходимо написать программу,
+# которая генерирует такую пирамиду со значением N, равным 10
+
+pyramid = int(input("Enter pyramid level: "))
+# for level in range(pyramid):
+for level in range(pyramid):
+    print(' ' * (pyramid-level-1) + '*' * (level*2+1))
+
+
+# # # Статуи # # #
+# Вы получили в подарок на день рождения статуи разных размеров,
+# каждая статуя имеет неотрицательный целочисленный размер.
+# Поскольку Вам нравится доводить вещи до совершенства,
+# то необходимо расположить их от меньшего к большему,
+# чтобы каждая статуя была больше предыдущей ровно на 1.
+# Для этого Вам могут понадобиться дополнительные статуи.
+# Определите количество отсутствующих статуй.
+# Пример
+# Для статуй = [6, 2, 3, 8] результат должен быть = 3.
+# Иными словами, у Вас отсутствуют статуи размеров 4, 5 и 7.
+
+gift = [6, 2, 3, 8]
+new_gift = list(range(min(gift), max(gift)+1))
+# print(new_gift)
```

Comment:
pls do not leave commented line in work repository

Reply:
fixed